### DISCUSSION TITLE
Is there a way to prevent auto analysis of some code regions/sections?

### DISCUSSION THREAD

**Discussion by ntqbit** [NationalSecurityAgency/ghidra#6198] [Q&A] [answered]
Hello.

I am trying to analyze a heavily obfuscated binary in Ghidra. However, due to the obfuscation (most likely vmprotect), that takes the majority of the binary code, Ghidra's auto analysis takes eternity to complete, while I do not even target those code regions since they rarely contain any references to non-obfuscated text section, plain function calls, anything that would be useful.

Is there a way to somehow speed up the analysis, for example by preventing analysis of some regions of code (from address - to address), or some sections?
It would save a lot of time.

Thanks!

**Comment by ryanmkurtz** [comment, marked as answer]
Have you tried going into the Memory Map and marking those blocks non-executable?
